### DISCUSSION TITLE
How to create a sitewide tree?

### DISCUSSION THREAD

**Discussion by Emmanuel-Messulam** [mkdocs/mkdocs#3077] [Q&A] [answered]
Thanks for the tool! It is really good.

I have no idea how to provide a sitewide tree in a kind of "index" page, because our organization docs are gaining a lot of submenues and files, and navigation can be a bit complex. Also it would allow you to search by topic more easily. Are there any plugins for this?

Thanks!

**Comment by oprypin** [comment, marked as answer]
You can manually create such a Markdown document that replicates your current nav, and then with https://github.com/oprypin/mkdocs-literate-nav you can use that document as the actual definition of your nav and delete the current definition of it.

**Reply by Emmanuel-Messulam** [reply]
You mean I should use a single SUMMARY.md as the index for [oprypin/mkdocs-literate-nav](https://github.com/oprypin/mkdocs-literate-nav) and as a page containing the tree?

**Reply by oprypin** [reply]
Yeah or it could be in any file name, not necessarily SUMMARY

**Reply by Emmanuel-Messulam** [reply]
Is there a way to generate a more *graphical* tree?

**Reply by pawamoy** [reply]
Something like https://github.com/markmap/mkdocs-markmap maybe?

**Reply by Emmanuel-Messulam** [reply]
> Something like [markmap/mkdocs-markmap](https://github.com/markmap/mkdocs-markmap) maybe?

Looks good! But it doesn't seem to auto update. It seems to be more catered to be a mind map.

**Reply by pawamoy** [reply]
If you want the autogenerate the tree with the current navigation, you'll a need a custom template and some Jinja markup (to access and iterate the different pages added to the nav) + some other markup like mermaid.js to render a tree of it

**Reply by Emmanuel-Messulam** [reply]
Makes sense, thanks!

**Comment by tombreit** [comment]
Perhaps something like https://tombreit.github.io/mkdocs-pagetree-plugin/ ?
(Shameless advertising: I am the author of the plugin)
